spec/unit/http/resolver_spec.rb

```diff
@@ -22,9 +22,50 @@
     end
   end
 
+  context 'when resolving using server_list' do
+    let(:server_list) { [["ca.example.com", "8141"], ["apple.example.com"]] }
+    let(:default_port) { '8142' }
+    let(:subject) { Puppet::HTTP::Resolver::ServerList.new(client, server_list: server_list, default_port: default_port) }
+
+    it 'yields a service based on the current server_list setting' do
```

Comment:
It looks like the tests are still passing a block to the `Resolver#resolve` method, but now it returns the service instead of yielding it. Could you update the tests in this file, eg `service = subject.resolve(session, :ca)`, update the test descriptions (eg `it 'returns a service`), and remove the `&block` argument from the various resolvers? Also could you add some verbiage to the first commit that introduced the server list resolver?

Reply:
done